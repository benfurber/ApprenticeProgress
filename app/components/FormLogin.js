// @flow

import React, { Component } from "react";
import { StyleSheet } from "react-native";
import { Content, Form, Input, Item, Label, View } from "native-base";

import { FormLoginButton } from "components";
import { colours, standards } from "styles";
import type { EmailType, navigationType, PasswordType } from "types";
import { presentError } from "utils";

type Props = {
  navigation: navigationType,
};

type State = {
  email: EmailType,
  password: PasswordType,
  error: ?String,
  isLoading: boolean,
};

class FormLogin extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      email: null,
      password: null,
      error: null,
      isLoading: false,
    };
  }

  componentDidUpdate(_, prevState) {
    const { error } = this.state;

    if (error !== prevState.error) {
      return presentError({ error });
    }
  }

  render() {
    const { navigation } = this.props;
    const { email, password } = this.state;

    return (
      <Content>
        <View style={styles.content}>
          <Form>
            <Item style={styles.formField} floatingLabel>
              <Label>Email</Label>
              <Input
                autoCapitalize="none"
                textContentType="emailAddress"
                onChangeText={email => this.setState({ email })}
                value={this.state.email}
              />
            </Item>
            <Item style={styles.formField} floatingLabel>
              <Label>Password</Label>
              <Input
                secureTextEntry={true}
                onChangeText={password => this.setState({ password })}
                value={this.state.password}
              />
            </Item>
            <FormLoginButton
              email={email}
              password={password}
              navigation={navigation}
              addError={error => this.setState({ error })}
              setLoadingState={boolean => this.setLoadingState(boolean)}
              isLoading={() => this.isLoading()}
            />
          </Form>
        </View>
      </Content>
    );
  }

  setLoadingState(boolean) {
    this.setState({ isLoading: boolean });
  }

  isLoading() {
    return this.state.isLoading;
  }
}

const styles = StyleSheet.create({
  content: {
    paddingTop: 200,
  },
  formField: {
    backgroundColor: colours.transparentWhite,
    ...standards.elementPadding,
  },
});

export { FormLogin };
