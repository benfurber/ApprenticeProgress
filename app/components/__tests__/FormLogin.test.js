// @flow

import React from "react";
import { shallow } from "enzyme";

import navigationMock from "mocks";

import { FormLogin } from "../FormLogin";

describe("Loading", () => {
  it("has a valid snapshot", () => {
    const component = shallow(
      <FormLogin
        navigation={navigationMock}
        email="testing@ben.com"
        password="YOMANCHANGEYOPASSWORD"
      />
    );

    expect(component).toMatchSnapshot();
  });
});
