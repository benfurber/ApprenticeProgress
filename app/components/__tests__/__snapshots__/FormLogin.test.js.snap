// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading has a valid snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormLogin />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(ViewNB)
        style={
          Object {
            "paddingTop": 200,
          }
        }
      >
        <Styled(Form)>
          <Styled(Item)
            floatingLabel={true}
            style={
              Object {
                "backgroundColor": "#FFFFFFCF",
                "paddingHorizontal": 30,
                "paddingVertical": 10,
              }
            }
          >
            <Styled(Label)>
              Email
            </Styled(Label)>
            <Styled(Input) />
          </Styled(Item)>
          <Styled(Item)
            floatingLabel={true}
            last={true}
            style={
              Object {
                "backgroundColor": "#FFFFFFCF",
                "paddingHorizontal": 30,
                "paddingVertical": 10,
              }
            }
          >
            <Styled(Label)>
              Password
            </Styled(Label)>
            <Styled(Input) />
          </Styled(Item)>
          <Apollo(FormLoginButtonUnwrapped)
            email="ben@fu.rber.io"
            password="temppasswordhere"
          />
        </Styled(Form)>
      </Styled(ViewNB)>,
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Form)>
          <Styled(Item)
            floatingLabel={true}
            style={
              Object {
                "backgroundColor": "#FFFFFFCF",
                "paddingHorizontal": 30,
                "paddingVertical": 10,
              }
            }
          >
            <Styled(Label)>
              Email
            </Styled(Label)>
            <Styled(Input) />
          </Styled(Item)>
          <Styled(Item)
            floatingLabel={true}
            last={true}
            style={
              Object {
                "backgroundColor": "#FFFFFFCF",
                "paddingHorizontal": 30,
                "paddingVertical": 10,
              }
            }
          >
            <Styled(Label)>
              Password
            </Styled(Label)>
            <Styled(Input) />
          </Styled(Item)>
          <Apollo(FormLoginButtonUnwrapped)
            email="ben@fu.rber.io"
            password="temppasswordhere"
          />
        </Styled(Form)>,
        "style": Object {
          "paddingTop": 200,
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Item)
              floatingLabel={true}
              style={
                Object {
                  "backgroundColor": "#FFFFFFCF",
                  "paddingHorizontal": 30,
                  "paddingVertical": 10,
                }
              }
            >
              <Styled(Label)>
                Email
              </Styled(Label)>
              <Styled(Input) />
            </Styled(Item)>,
            <Styled(Item)
              floatingLabel={true}
              last={true}
              style={
                Object {
                  "backgroundColor": "#FFFFFFCF",
                  "paddingHorizontal": 30,
                  "paddingVertical": 10,
                }
              }
            >
              <Styled(Label)>
                Password
              </Styled(Label)>
              <Styled(Input) />
            </Styled(Item)>,
            <Apollo(FormLoginButtonUnwrapped)
              email="ben@fu.rber.io"
              password="temppasswordhere"
            />,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Styled(Label)>
                  Email
                </Styled(Label)>,
                <Styled(Input) />,
              ],
              "floatingLabel": true,
              "style": Object {
                "backgroundColor": "#FFFFFFCF",
                "paddingHorizontal": 30,
                "paddingVertical": 10,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Email",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": "Email",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Styled(Label)>
                  Password
                </Styled(Label)>,
                <Styled(Input) />,
              ],
              "floatingLabel": true,
              "last": true,
              "style": Object {
                "backgroundColor": "#FFFFFFCF",
                "paddingHorizontal": 30,
                "paddingVertical": 10,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Password",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": "Password",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "email": "ben@fu.rber.io",
              "password": "temppasswordhere",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(ViewNB)
          style={
            Object {
              "paddingTop": 200,
            }
          }
        >
          <Styled(Form)>
            <Styled(Item)
              floatingLabel={true}
              style={
                Object {
                  "backgroundColor": "#FFFFFFCF",
                  "paddingHorizontal": 30,
                  "paddingVertical": 10,
                }
              }
            >
              <Styled(Label)>
                Email
              </Styled(Label)>
              <Styled(Input) />
            </Styled(Item)>
            <Styled(Item)
              floatingLabel={true}
              last={true}
              style={
                Object {
                  "backgroundColor": "#FFFFFFCF",
                  "paddingHorizontal": 30,
                  "paddingVertical": 10,
                }
              }
            >
              <Styled(Label)>
                Password
              </Styled(Label)>
              <Styled(Input) />
            </Styled(Item)>
            <Apollo(FormLoginButtonUnwrapped)
              email="ben@fu.rber.io"
              password="temppasswordhere"
            />
          </Styled(Form)>
        </Styled(ViewNB)>,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Form)>
            <Styled(Item)
              floatingLabel={true}
              style={
                Object {
                  "backgroundColor": "#FFFFFFCF",
                  "paddingHorizontal": 30,
                  "paddingVertical": 10,
                }
              }
            >
              <Styled(Label)>
                Email
              </Styled(Label)>
              <Styled(Input) />
            </Styled(Item)>
            <Styled(Item)
              floatingLabel={true}
              last={true}
              style={
                Object {
                  "backgroundColor": "#FFFFFFCF",
                  "paddingHorizontal": 30,
                  "paddingVertical": 10,
                }
              }
            >
              <Styled(Label)>
                Password
              </Styled(Label)>
              <Styled(Input) />
            </Styled(Item)>
            <Apollo(FormLoginButtonUnwrapped)
              email="ben@fu.rber.io"
              password="temppasswordhere"
            />
          </Styled(Form)>,
          "style": Object {
            "paddingTop": 200,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Item)
                floatingLabel={true}
                style={
                  Object {
                    "backgroundColor": "#FFFFFFCF",
                    "paddingHorizontal": 30,
                    "paddingVertical": 10,
                  }
                }
              >
                <Styled(Label)>
                  Email
                </Styled(Label)>
                <Styled(Input) />
              </Styled(Item)>,
              <Styled(Item)
                floatingLabel={true}
                last={true}
                style={
                  Object {
                    "backgroundColor": "#FFFFFFCF",
                    "paddingHorizontal": 30,
                    "paddingVertical": 10,
                  }
                }
              >
                <Styled(Label)>
                  Password
                </Styled(Label)>
                <Styled(Input) />
              </Styled(Item)>,
              <Apollo(FormLoginButtonUnwrapped)
                email="ben@fu.rber.io"
                password="temppasswordhere"
              />,
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Styled(Label)>
                    Email
                  </Styled(Label)>,
                  <Styled(Input) />,
                ],
                "floatingLabel": true,
                "style": Object {
                  "backgroundColor": "#FFFFFFCF",
                  "paddingHorizontal": 30,
                  "paddingVertical": 10,
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Email",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": "Email",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Styled(Label)>
                    Password
                  </Styled(Label)>,
                  <Styled(Input) />,
                ],
                "floatingLabel": true,
                "last": true,
                "style": Object {
                  "backgroundColor": "#FFFFFFCF",
                  "paddingHorizontal": 30,
                  "paddingVertical": 10,
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Password",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": "Password",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "email": "ben@fu.rber.io",
                "password": "temppasswordhere",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
